### PR TITLE
Backbone.Relational does not work with require.js

### PR DESCRIPTION
**Pull request by peters**
Use factory pattern so that it also works with require.js

### CODE REVIEW

**Comment by dowglaz** [comment]
Hello guys.

I know this discussion is old, but I faced some terrible hours trying to figure out why including this library on my project was causing it to completely break...

I'm using Backbone and requirejs on my project and I was making the shim workaround and nothing was working. So I looked into the source code of backbone-relational and noticed that there's two explicit requires of 'backbone' and 'underscore' on line 51... the problem is that I'm using my libraries in 'lib/backbone' and 'lib/underscore' and even declaring the paths on requirejs.config it didn't work.

In the end I found a configuration that just works that is:
```js
requirejs.config({
  //...
  paths: {
    'underscore': 'lib/underscore',
    'backbone': 'lib/backbone'
  },
  shim: {
    'lib/underscore': {
      deps: ['lib/jquery'],
      exports: '_'
    },
    'lib/backbone': {
      deps: ['lib/underscore', 'lib/jquery'],
      exports: 'Backbone'
    },
    'backbone-relational': ['backbone'],
  }
});
```
And also force requiring it without the "lib" prefix (because require 'lib/backbone' and 'lib/underscore' doesnt' work):
```js
define([
    'lib/jquery',
    'backbone',
    'underscore',
    'backbone-relational'
  ],
  function ($, backbone, _) {
    return backbone;
  }
);
```

In other words, I was forced to change somethings on my project (worst, I had to found this practically at random, since I was not understanding what was happening .. I know it's my fault also, but I'm kinda newbie on JS and requirejs :P) and I noticed that there's a possible solution, at least for requirejs: just remove the lines 50-52 may work since the programmer make a shim exporting `'backbone'` to `window.Backbone` and `'underscore'` to `window._` .

It's less obtrusive and don't force the programmer to make a lot of changes on the project just because one dependency. And just works, regardless of whether there's another export of `'backbone'` to any other variable, nothing blocks you from making more than one shim.

**Comment by philfreo** [comment]
@dowglaz I think you're over complicating things. You shouldn't need shims / you shouldn't need to define the dependencies of backbone explicitly if you're using the AMD versions of everything which work well together.

My require js config is pretty simple:

    paths: {
        'underscore': 'vendor/underscore',
        'backbone': 'vendor/backbone/backbone',
        'backbone-relational': 'vendor/backbone-relational/backbone-relational',
    }


And I have one additional file/module called 'backbone-all' that I use & require anywhere that I expect to load backbone with all my backbone plugins:

```
define([
    'underscore',
    'backbone',
    'backbone-relational',
],

function(Backbone) {
    // Put any custom Backbone behavior here

    return Backbone;
});
```
